### PR TITLE
BUG: Don't print stray newline with MultiIndex

### LINKED ISSUE #6618: BUG: to_csv extra header line with multiindex columns
This seems strange to me, but I don't often use a MultiIndex so I might be missing something obvious.

```
>>> pd.__version__
'0.13.1-420-g6899ed6'
>>> df2 = pd.DataFrame([1], columns=pd.MultiIndex.from_arrays([[1],[2]]))
>>> df2
   1
   2
0  1

[1 rows x 1 columns]
>>> df2.columns
MultiIndex(levels=[[1], [2]],
           labels=[[0], [0]])
>>> print df2.to_csv()
,1
,2
,
0,1
```

Is there supposed to be that empty line at the end of the header?  Compare

```
>>> print df2.to_csv(header=False)
0,1
```

### PR DESCRIPTION
**Pull request by gfyoung**
Title is self-explanatory.

Closes #6618.

### CODE REVIEW

File: pandas/tests/formats/test_format.py [hunk 1]
```diff
@@ -3327,6 +3327,16 @@ def test_to_csv_date_format(self):
         self.assertEqual(df_sec_grouped.mean().to_csv(date_format='%Y-%m-%d'),
                          expected_ymd_sec)
 
+    def test_to_csv_multi_index(self):
+        # see gh-6618
+        df = DataFrame([1], columns=pd.MultiIndex.from_arrays([[1],[2]]))
+
+        exp = ",1\n,2\n,\n0,1\n"
+        self.assertEqual(df.to_csv(), exp)
```
Comment (jreback) [inline pandas/tests/formats/test_format.py]: test with mi names as well

File: pandas/formats/format.py [hunk 2]
```diff
@@ -1547,12 +1547,11 @@ def _save_header(self):
...
                 writer.writerow(col_line)
 
-            # add blanks for the columns, so that we
-            # have consistent seps
-            encoded_labels.extend([''] * len(columns))
-
-        # write out the index label line
-        writer.writerow(encoded_labels)
+            # Write out the index line if it's not empty.
+            # Otherwise, we have this awkward blank line.
```
Comment (jreback) [inline pandas/formats/format.py]: more useful comment - this only make sense if u know about the issue

File: pandas/tests/formats/test_format.py (hunk 1 above)
Comment (gfyoung) [inline pandas/tests/formats/test_format.py]: Yep, done.

File: pandas/formats/format.py (hunk 2 above)
Comment (gfyoung) [inline pandas/formats/format.py]: Yep, done.

**Comment by jreback** [comment]
thanks!

**Pull request closed without merging**